player/command.c

```diff
@@ -1997,6 +2000,9 @@ static int get_track_entry(int item, int action, void *arg, void *ctx)
     struct MPContext *mpctx = ctx;
     struct track *track = mpctx->tracks[item];
 
+    char *external_filename = mp_normalize_user_path(NULL, mpctx->global,
+                                                     track->external_filename);
```

Comment:
1. this keeps spamming `[   1.966][d][global] user path: 'folder.jpg' -> 'folder.jpg'` in my logs. depends on circumstances `track-list` may (apparently) be read a few times per osc update.
 obviously the name should only be normalized once
2. why does this go through "user_path" stuff at all? it's not like the external filename is supposed to contain e.g. `~~global/` or so

Reply:
1. mp_get_user_path logs everytime it's called. Anytime the property is read, this will get called. One way to avoid that would be to save the track's external filename as expanded directly, but not sure if that would have some other unforeseen consequence or so.
2. It needs it in order to expand `~/path` which works now. `mp_normalize_path` on its own is too dumb to handle that.

Reply:
> One way to avoid that would be to save the track's external filename as expanded directly, but not sure if that would have some other unforeseen consequence or so.

Well, normalizing the filename once when you open it and remembering what the path was sounds like the correct behavior to me.